crates/lsp/src/lsp.rs

```diff
@@ -215,6 +219,19 @@ impl LanguageServer {
             &binary.arguments
         );
 
+        #[cfg(target_os = "windows")]
+        let mut server = process::Command::new(&binary.path)
+            .current_dir(working_dir)
+            .creation_flags(0x08000000) // CREATE_NO_WINDOW
+            .args(binary.arguments)
+            .envs(binary.env.unwrap_or_default())
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .kill_on_drop(true)
+            .spawn()?;
```

Comment:
I prefer below:

```suggestion
        let mut command = process::Command::new(&binary.path);
        command 
            .current_dir(working_dir)
            .args(binary.arguments)
            .envs(binary.env.unwrap_or_default())
            .stdin(Stdio::piped())
            .stdout(Stdio::piped())
            .stderr(Stdio::piped())
            .kill_on_drop(true);
        #[cfg(windows)]
        command.creation_flags(windows::Win32::System::Threading::CREATE_NO_WINDOW.0);
        let mut server = command.spawn()?;
```

crates/lsp/Cargo.toml
```toml
[target.'cfg(windows)'.dependencies]
windows.workspace = true
```